pkg/drivers/kvm/network.go

```diff
@@ -99,8 +103,24 @@ func (d *Driver) ensureNetwork() error {
 
 	// Start the private network
 	log.Infof("Ensuring network %s is active", d.PrivateNetwork)
+	// retry once to recreate the network, but only if is not used by another minikube instance
 	if err := setupNetwork(conn, d.PrivateNetwork); err != nil {
-		return err
+		log.Debugf("Network %s is inoperable, will try to recreate it: %v", d.PrivateNetwork, err)
+		if err := d.deleteNetwork(); err != nil {
+			return errors.Wrapf(err, "deleting inoperable network %s", d.PrivateNetwork)
+		}
+		log.Debugf("Successfully deleted %s network", d.PrivateNetwork)
+		time.Sleep(nap)
+		if err := d.createNetwork(); err != nil {
+			return errors.Wrapf(err, "recreating inoperable network %s", d.PrivateNetwork)
+		}
+		log.Debugf("Successfully recreated %s network", d.PrivateNetwork)
+		time.Sleep(nap)
+		if err := setupNetwork(conn, d.PrivateNetwork); err != nil {
+			return err
+		}
+		log.Debugf("Successfully activated %s network", d.PrivateNetwork)
+		time.Sleep(nap)
```

Comment:
is there a way we can verify libvirt ops went though by verifying ? instead of waiting ?
(we could use one of our rerry package here, for example retry.Expo... (a fun that checks if it went through) up to 10 seconds...

Reply:
Agree with @medyagh , a solution that requires no sleeps would be optimal.

Reply:
@medyagh , @priyawadhwa thanks for review & comments - good point, agree and i've amended pr accordingly: please have a look